core/io/resource.h

```diff
@@ -103,6 +103,7 @@ class Resource : public RefCounted {
 
 	virtual void set_path(const String &p_path, bool p_take_over = false);
 	String get_path() const;
+	void set_temporary_path(const String &p_path);
```

Comment:
Have not looked in depth but since its hacky, maybe better to use a metadata, then clear it?

Reply:
I would name it for what it does. 
```suggestion
	void set_path_cache(const String &p_path);
```